tachyon/zk/plonk/vanishing/vanishing_evaluated.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_VANISHING_VANISHING_EVALUATED_H_
+#define TACHYON_ZK_PLONK_VANISHING_VANISHING_EVALUATED_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/zk/base/entities/entity_ty.h"
+#include "tachyon/zk/plonk/vanishing/vanishing_committed.h"
+
+namespace tachyon::zk {
+
+template <EntityTy EntityType, typename PCSTy>
+class VanishingEvaluated;
+
+template <typename PCSTy>
+class VanishingEvaluated<EntityTy::kProver, PCSTy> {
+ public:
+  using F = typename PCSTy::Field;
+  using Poly = typename PCSTy::Poly;
+
+  VanishingEvaluated() = default;
+  VanishingEvaluated(Poly&& h_poly, F&& h_blinds_poly_eval,
+                     VanishingCommitted<EntityTy::kProver, PCSTy>&& committed)
+      : h_poly_(std::move(h_poly)),
+        h_blinds_poly_eval_(std::move(h_blinds_poly_eval)),
+        committed_(std::move(committed)) {}
+
+  Poly&& TakeHPoly() && { return std::move(h_poly_); }
+  F&& TakeHBlindsPolyEval() && { return std::move(h_blinds_poly_eval_); }
+  VanishingCommitted<EntityTy::kProver, PCSTy>&& TakeCommitted() && {
+    return std::move(committed_);
+  }
+
+ private:
+  Poly h_poly_;
+  F h_blinds_poly_eval_;
```

Comment:
What is the reason for naming this variable h_blinds_poly_eval_ instead of h_blind as in the original code

Reply:
To make [this](https://github.com/kroma-network/halo2/blob/7d0a36990452c8e7ebd600de258420781a9b7917/halo2_proofs/src/plonk/vanishing/prover.rs#L138-L144) code readable, I've refactored the code as an evaluation of a h_blinds_poly on x_n. So I named it `h_blinds_poly_eval_`.